src/garrysmod/gamemodes/zombiesurvival/gamemode/howl/sv_howl.lua

```diff
@@ -0,0 +1,261 @@
+Howl = {}
+Howl.logger = include("lib/log.lua")
+Howl.commands = {}
+Howl.roles = {
+    ["admin"] = {
+        root = true,
+        -- These are skipped due to root and are for illustrative purposes only.
+        perms = { "administrate", "moderate" }
+    },
+    ["moderator"] = {
+        perms = { "moderate" }
+    },
+    ["user"] = {
+        perms = {}
+    }
+}
+
+-- Metatable overrides
+AddCSLuaFile("meta/cl_player.lua")
+AddCSLuaFile("meta/sh_player.lua")
+include("meta/sh_player.lua")
+include("meta/sv_player.lua")
+
+-- Howl
+AddCSLuaFile("sh_util.lua")
+include("sh_util.lua")
+include("sv_hooks.lua")
+include("sv_command.lua")
+
+function Howl:register_command(command)
+    for k, v in pairs(self.commands) do
+        if v.name == command.name then
+            table.remove(self.commands, k)
+        end
+    end
+
+    table.insert(self.commands, command)
+end
+
+function Howl:find_command(name)
+    for _, command in ipairs(self.commands) do
+        if name == command.name then
+            return command
+        end
+    end
+end
+
+function Howl:available_commands_message(pl)
+    local available_commands = {}
+
+    for _, command in ipairs(self.commands) do
+        if command ~= self and (not pl:IsValid() or pl:can(command.perm)) then
+            table.insert(available_commands, "• " .. string.Split(command:usage(), "usage: ")[2])
+        end
+    end
+
+    return string.format(
+        "AVAILABLE COMMANDS\n%s",
+        table.concat(available_commands, "\n")
+    )
+end
+
+function Howl:extract_arguments(text)
```

Comment:
Borrowed from Flux: https://github.com/TeslaCloud/flux-ce/blob/d5cf68d60c8db0ea40540786e0728a3fb42a1998/crates/flow/lib/sh_command.lua#L84.

Rather than simply splitting the string by spaces to generate the array of arguments, this function also checks for quoting.